extension/src/time-sheets/summary.ts

```diff
@@ -81,23 +74,21 @@ const determineTotalDaysSavedFromTimeCards = (timeSheet: TimeSheet): number => {
   return totalDaysSaved
 }
 
-const determineNextDueDate = (timeSheet: TimeSheet, alertableLastDayOfMonth: DayOfWeek | null): Date => {
-  const deadlineHour = 17 // 5pm
+const determineNextDueDate = (
+  timeSheet: TimeSheet,
+  extensionOptions: ExtensionOptions,
+  alertableLastDayOfMonth: DateOnly | null,
+): Date => {
   const now = new Date()
+  const deadlineTime = extensionOptions.weeklyDueTime
 
   if (alertableLastDayOfMonth) {
-    const todayDateOnly = getTodayDateOnly()
-    const todayDayOfWeek = getDayOfWeek(todayDateOnly)
-    const todayDayOfWeekIndex = dayOfWeekIndexes[todayDayOfWeek]
-    const alertableLastDayOfMonthIndex = dayOfWeekIndexes[alertableLastDayOfMonth]
-    const alertableDateOnly = addDays(todayDateOnly, alertableLastDayOfMonthIndex - todayDayOfWeekIndex)
-
     const alertableDueDate = new Date(
```

Comment:
I probably should make something to construct a JavaScript `Date` from `DateOnly` and `TimeOnly` combined.